Intel compiler error

Trying tag 20.09 with 
which mpif90
/soft/apps/packages/mpich-3.2-intel17-initial/bin/mpif90
...
make  -f GNUmakefile_intel_1D                                                                                                                              
Loading /nfs/proj-flash5/amrex_testing_clone/Tools/GNUMake/comps/intel.mak...
Loading /nfs/proj-flash5/amrex_testing_clone/Tools/GNUMake/sites/Make.unknown...
Compiling AMReX_multifab_mod.F90 ...
mpif90  -g -O2 -ip -qopt-report=5 -qopt-report-phase=vec -implicitnone   -module tmp_build_dir/o/1d.intel.MPI.EXE -Itmp_build_dir/o/1d.intel.MPI.EXE -DBL_LANG_FORT -DAMREX_LANG_FORT -DBL_USE_MPI -DAMREX_USE_MPI -DAMREX_GIT_VERSION=\"20.09\" -DAMREX_LAUNCH="" -DAMREX_DEVICE="" -DAMREX_CUDA_FORT_GLOBAL="" -DAMREX_CUDA_FORT_DEVICE="" -DAMREX_CUDA_FORT_HOST="" -DAMREX_CUDA_FORT_HOST_DEVICE="" -DBL_SPACEDIM=1 -DAMREX_SPACEDIM=1 -DBL_FORT_USE_UNDERSCORE -DAMREX_FORT_USE_UNDERSCORE -DBL_Linux -DAMREX_Linux -DAMREX_PARTICLES -DNDEBUG -DAMREX_PARTICLES -DBL_USE_F_INTERFACES -I. -I/nfs/proj-flash5/amrex_testing_clone/Src/Base -I/nfs/proj-flash5/amrex_testing_clone/Src/AmrCore -I/nfs/proj-flash5/amrex_testing_clone/Src/Amr -I/nfs/proj-flash5/amrex_testing_clone/Src/Boundary -I/nfs/proj-flash5/amrex_testing_clone/Src/Particle -I/nfs/proj-flash5/amrex_testing_clone/Src/F_Interfaces/Particle -I/nfs/proj-flash5/amrex_testing_clone/Src/F_Interfaces/Base -I/nfs/proj-flash5/amrex_testing_clone/Src/F_Interfaces/Octree -I/nfs/proj-flash5/amrex_testing_clone/Src/F_Interfaces/AmrCore -I/nfs/proj-flash5/amrex_testing_clone/Src/LinearSolvers/MLMG -I/nfs/proj-flash5/amrex_testing_clone/Src/LinearSolvers/Projections -I/nfs/proj-flash5/amrex_testing_clone/Src/F_Interfaces/LinearSolvers -c /nfs/proj-flash5/amrex_testing_clone/Src/F_Interfaces/Base/AMReX_multifab_mod.F90 -o tmp_build_dir/o/1d.intel.MPI.EXE/AMReX_multifab_mod.o
ifort: remark #10397: optimization reports are generated in *.optrpt files in the output location
/nfs/proj-flash5/amrex_testing_clone/Src/F_Interfaces/Base/AMReX_multifab_mod.F90(47): error #8259: The type bound procedure definition statement must contains only one binding name.   [AMREX_MULTIFAB_SETVAL_GV]
     procedure :: setval        => amrex_multifab_setval, amrex_multifab_setval_gv
----------------------------------------------------------^
/nfs/proj-flash5/amrex_testing_clone/Src/F_Interfaces/Base/AMReX_multifab_mod.F90(56): error #8259: The type bound procedure definition statement must contains only one binding name.   [AMREX_MULTIFAB_COPY_CGV]
     procedure :: copy          => amrex_multifab_copy, amrex_multifab_copy_cgv ! This copies the data
--------------------------------------------------------^
/nfs/proj-flash5/amrex_testing_clone/Src/F_Interfaces/Base/AMReX_multifab_mod.F90(102): error #8259: The type bound procedure definition statement must contains only one binding name.   [AMREX_IMULTIFAB_SETVAL_GV]
     procedure :: setval        => amrex_imultifab_setval, amrex_imultifab_setval_gv
-----------------------------------------------------------^
compilation aborted for /nfs/proj-flash5/amrex_testing_clone/Src/F_Interfaces/Base/AMReX_multifab_mod.F90 (code 1)
make: *** [tmp_build_dir/o/1d.intel.MPI.EXE/AMReX_multifab_mod.o] Error 1

It might be that I need to update my intel compiler based on: https://community.intel.com/t5/Intel-Fortran-Compiler/bug-error-8259-TBP-def-must-contain-1-binding/td-p/1047332

--- a/Src/F_Interfaces/Base/AMReX_multifab_mod.F90
+++ b/Src/F_Interfaces/Base/AMReX_multifab_mod.F90
@@ -44,7 +44,7 @@ module amrex_multifab_module
      procedure :: norm0         => amrex_multifab_norm0
      procedure :: norm1         => amrex_multifab_norm1
      procedure :: norm2         => amrex_multifab_norm2
-     procedure :: setval        => amrex_multifab_setval, amrex_multifab_setval_gv
+     generic :: setval        => amrex_multifab_setval, amrex_multifab_setval_gv
      procedure :: plus          => amrex_multifab_plus
      procedure :: mult          => amrex_multifab_mult
      procedure :: add           => amrex_multifab_add
@@ -53,13 +53,17 @@ module amrex_multifab_module
      procedure :: divide        => amrex_multifab_divide
      procedure :: saxpy         => amrex_multifab_saxpy
      procedure :: lincomb       => amrex_multifab_lincomb
-     procedure :: copy          => amrex_multifab_copy, amrex_multifab_copy_cgv ! This copies the data
+     generic :: copy          => amrex_multifab_copy, amrex_multifab_copy_cgv ! This copies the data
      generic   :: parallel_copy => amrex_multifab_parallel_copy, amrex_multifab_parallel_copy_c, &
           amrex_multifab_parallel_copy_cg, amrex_multifab_parallel_copy_cgv
      generic   :: fill_boundary => amrex_multifab_fill_boundary, amrex_multifab_fill_boundary_c
      generic   :: override_sync => amrex_multifab_override_sync, amrex_multifab_override_sync_mask
      generic   :: sum_boundary  => amrex_multifab_sum_boundary, amrex_multifab_sum_boundary_c
      procedure :: average_sync  => amrex_multifab_average_sync
+     procedure, private :: amrex_multifab_setval
+     procedure, private :: amrex_multifab_setval_gv
+     procedure, private :: amrex_multifab_copy
+     procedure, private :: amrex_multifab_copy_cgv
      procedure, private :: amrex_multifab_fill_boundary
      procedure, private :: amrex_multifab_fill_boundary_c
      procedure, private :: amrex_multifab_parallel_copy
@@ -99,8 +103,10 @@ module amrex_multifab_module
      procedure :: nghostvect    => amrex_imultifab_nghost_vect
      procedure :: nodal_type    => amrex_imultifab_nodal_type ! get index type
      procedure :: dataPtr       => amrex_imultifab_dataptr
-     procedure :: setval        => amrex_imultifab_setval, amrex_imultifab_setval_gv
+     generic   :: setval        => amrex_imultifab_setval, amrex_imultifab_setval_gv
      procedure, private :: amrex_imultifab_assign
+     procedure, private :: amrex_imultifab_setval_gv
+     procedure, private :: amrex_imultifab_setval
 #if !defined(__GFORTRAN__) || (__GNUC__ > 4)
      final :: amrex_imultifab_destroy
 #endif

